File: backend/models/artifact.py
from pydantic import BaseModel

from backend.models.character import CharacterClass, Quality


class ArtifactEffect(BaseModel):
    level: int
    description: str


class ArtifactTreasure(BaseModel):
    name: str
    lore: str
    character_class: CharacterClass
    effect: list[ArtifactEffect]


class Artifact(BaseModel):
    name: str
    is_global: bool
    lore: str
    quality: Quality
    effect: list[ArtifactEffect]
    width: int
    height: int
    treasures: list[ArtifactTreasure]

File: frontend/src/types/artifact.ts
import type { CharacterClass } from './character';
import type { Quality } from './quality';

export interface ArtifactEffect {
  level: number;
  description: string;
}

export interface ArtifactTreasure {
  name: string;
  lore: string;
  character_class: CharacterClass;
  effect: ArtifactEffect[];
}

export interface Artifact {
  name: string;
  is_global: boolean;
  lore: string;
  quality: Quality;
  effect: ArtifactEffect[];
  columns: number;
  rows: number;
  treasures: ArtifactTreasure[];
  last_updated: number;
}
